### PR TITLE
vpp_unittest: implement unittest for ProAmp

### PR DESCRIPTION
**Pull request by wudping**
mplement unittest for ProAmp, testing hue, saturation, brightness
and contrast.

unittest for #629
Signed-off-by: wudping <dongpingx.wu@intel.com>

### CODE REVIEW

File: vpp/vaapipostprocess_scaler_unittest.cpp [hunk 1]
```diff
@@ -255,4 +255,36 @@ VAAPIPOSTPROCESS_SCALER_TEST(Deinterlace)
...
+    EXPECT_EQ(YAMI_SUCCESS, scaler.setParameters(VppParamTypeColorBalance, &params));
+    EXPECT_EQ(YAMI_SUCCESS, scaler.process(src, dest));
+
+    params.mode = COLORBALANCE_CONTRAST;
+    params.level = 50;
+    EXPECT_EQ(YAMI_SUCCESS, scaler.setParameters(VppParamTypeColorBalance, &params));
+    EXPECT_EQ(YAMI_SUCCESS, scaler.process(src, dest));
+}
+
+VAAPIPOSTPROCESS_SCALER_TEST(Clrbalance)
```
Comment (uartie) [inline vpp/vaapipostprocess_scaler_unittest.cpp]: Can you rename this test "ColorBalance" instead... the "Clr" part is ambiguous.  For test names, verbosity is better.

File: vpp/vaapipostprocess_scaler_unittest.cpp [hunk 2]
```diff
@@ -255,4 +255,36 @@ VAAPIPOSTPROCESS_SCALER_TEST(Deinterlace)
...
+
+    params.mode = COLORBALANCE_BRIGHTNESS;
+    params.level = 50;
+    EXPECT_EQ(YAMI_SUCCESS, scaler.setParameters(VppParamTypeColorBalance, &params));
+    EXPECT_EQ(YAMI_SUCCESS, scaler.process(src, dest));
+
+    params.mode = COLORBALANCE_CONTRAST;
+    params.level = 50;
+    EXPECT_EQ(YAMI_SUCCESS, scaler.setParameters(VppParamTypeColorBalance, &params));
+    EXPECT_EQ(YAMI_SUCCESS, scaler.process(src, dest));
```
Comment (uartie) [inline vpp/vaapipostprocess_scaler_unittest.cpp]: Should check negative path results, too (i.e. YAMI_INVALID_PARAM, YAMI_UNSUPPORTED, ...).  For example, "invalid param size", "invalid param mode", etc.

Also, do checks on m_colorBalance to verify params are actually honored/stored.

File: vpp/vaapipostprocess_scaler_unittest.cpp (hunk 1 above)
Comment (wudping) [reply vpp/vaapipostprocess_scaler_unittest.cpp]: okay, thanks @uartie .

File: vpp/vaapipostprocess_scaler_unittest.cpp (hunk 2 above)
Comment (wudping) [reply vpp/vaapipostprocess_scaler_unittest.cpp]: okay, thanks @uartie for your details. I will modify.

**Comment by wudping** [comment]
updated according comments, please help review.

**Comment by wudping** [comment]
this patch is based on [pull-request 681](https://github.com/01org/libyami/pull/681)

**Review by uartie — COMMENTED** [review]
Better. Thanks. +1

**Comment by seanvk** [comment]
lgtm +1

**Pull request merged**